cmd/gen-go-sample/out.go

```diff
@@ -270,8 +285,8 @@ func writeComment(cmtFmt string, cmtArgs []string, gen *generator) error {
 	buf.WriteString("\n")
 	prependLines(&buf, "// ", false)
 	cmts := strings.Split(buf.String(), "\n")
-	for i, c := range(cmts) {
-		if i == len(cmts) - 1 {
+	for i, c := range cmts {
+		if i == len(cmts)-1 {
```

Comment:
So this skips the last `cmt` item? maybe do `range cmts[:len(cmts)-1]` instead, or 
```go
end := len(cmts)-1
for _, c := range cmts[:end] {
  gen.pt.Printf(c)
}
```
Or some variation of that. Then you don't need `i` and it's a little more clear.

Reply:
Done.

BTW, is there a more elegant way to skip the last empty string when `strings.Split("a\nb\nc\n", "\n")?

Reply:
I'm not sure if this is objectively better but I like it a little better: https://play.golang.org/p/MtudlnG2OE8

Reply:
@pongad 's suggestion is certainly more readable. It's very clear that the code drops the last newline and goes about it's business, rather than dropping the last element  as I had suggested.